src/uct/tcp/tcp_listener.c

```diff
@@ -0,0 +1,216 @@
+/**
+* Copyright (C) Mellanox Technologies Ltd. 2019.  ALL RIGHTS RESERVED.
+*
+* See file LICENSE for terms.
+*/
+
+#include "tcp_listener.h"
+#include <ucs/sys/sock.h>
+#include <ucs/async/async.h>
+
+
+static void uct_tcp_listener_client_data_handler(int fd, void *arg)
+{
+}
+
+static void uct_tcp_listener_conn_req_handler(int fd, void *arg)
+{
+    uct_tcp_listener_t *listener = (uct_tcp_listener_t *)arg;
+    char ip_port_str[UCS_SOCKADDR_STRING_LEN];
+    struct sockaddr_storage client_addr;
+    ucs_async_context_t *async_ctx;
+    uct_tcp_listener_ctx_t *listen_ctx;
+    ucs_status_t status;
+    socklen_t addrlen;
+    int accept_fd;
+
+    ucs_assert(fd == listener->listener_fd);
+
+    for (;;) {
+        addrlen = sizeof(struct sockaddr_storage);
+
+        accept_fd = accept(listener->listener_fd, (struct sockaddr*)&client_addr,
+                           &addrlen);
+        if (accept_fd == -1) {
+            if ((errno != EAGAIN) && (errno != EWOULDBLOCK) && (errno != EINTR)) {
```

Comment:
can be optimized a bit like 
```
if ((errno == EAGAIN) || (errno == EWOULDBLOCK) || (errno == EINTR)) { /* <-- no need to check all errors every time */
    return;
}
ucs_error();
return;
```

Reply:
why is this better? we can have one return instead of two.

Reply:
we have the same if here:
https://github.com/openucx/ucx/blob/70c2513dba4049e5e6d7919ed49494153eb7b9f4/src/ucs/sys/sock.c#L231

can it be united to a utility function in UCS/SYS/SOCK that could be used in UCS and UCT?

Reply:
there is a chance do 1 check if errno == EAGAIN, or 3 every time as it's implemented now

Reply:
@dmitrygx how about instead i create a common ucs_sockect_accept function in sock.c and use it from both tcp_iface and tcp_listener?

Reply:
> @dmitrygx how about instead i create a common ucs_sockect_accept function in sock.c and use it from both tcp_iface and tcp_listener?

yes, it makes sense to me
but also would be great to have a separate function (declared as `static` - we will use it inside `ucs/sys/sock.c` only) that checks
`((errno != EAGAIN) && (errno != EWOULDBLOCK) && (errno != EINTR))` in order to decrease code duplication (use this utility function in `ucs_sockect_accept()` and `ucs_socket_handle_io_error()`)